core/src/main/java/com/redhat/lightblue/client/request/data/DataDeleteRequest.java

```diff
@@ -31,6 +31,7 @@ public JsonNode getBodyJson() {
         if (queryExpression != null) {
             node.set("query", queryExpression.toJson());
         }
+        appendExecutionToJson(node);
```

Comment:
> Is there a case an array will be used? I think it is all objects.

You're right, I checked incorrectly. Made the change. Didn't touch range, implementing it in a similar fashion would require one more abstract layer (because delete request does not offer range api) and it does not look like improvement over what we have.